test/con_cache_test.exs

```diff
@@ -184,6 +184,24 @@ defmodule ConCacheTest do
     end)
   end
 
+  test "get_or_store on bag" do
+    with_cache([ets_options: [:bag]],fn(cache) ->
+      assert ConCache.get_or_store(cache, :a, fn() -> 1 end) == 1
```

Comment:
For this pull request it will validate `get_or_store` the same way it does with `update_*`, i didn't look at the what needs to be changed, but I'll do it. Correct readme and docs. :+1: 

We could open an issue to have this discussion about the better way to act with `get_or_store`(I would like to do the same with `update_*` but different issues, since they are different scopes).